File: csharp/DVSClient/DVSClient/server/address/search/RestApiAddressSearchResult.cs
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Search
{
    public class RestApiAddressSearchResult
    {
        [JsonProperty("more_results_available")]
        public bool? MoreResultsAvailable { get; set; }

        [JsonProperty("confidence")]
        public string? Confidence { get; set; }

        [JsonProperty("suggestions_key")]
        public string? SuggestionsKey { get; set; }

        [JsonProperty("suggestions_prompt")]
        public string? SuggestionsPrompt { get; set; }

        [JsonProperty("suggestions")]
        public IEnumerable<RestApiSearchResultSuggestion>? Suggestions { get; set; }

        public bool? IsMoreResultsAvailable()
        {
            return MoreResultsAvailable;
        }
    }
}

File: typescript/src/server/address/search/restApiAddressSearchResult.ts
import { RestApiSearchResultSuggestion } from "./restApiSearchResultSuggestion";

export type RestApiAddressSearchResult = {
    more_results_available?: boolean;
    confidence?: string;
    suggestions_key?: string;
    suggestions_prompt?: string;
    suggestions?: RestApiSearchResultSuggestion[]
}